lib/adb-device.js

```diff
@@ -0,0 +1,38 @@
+import { CONNECTION_TYPES } from './constants';
+import USBDevice from './usb-device';
+
+// create one of these to interface with a device
+class ADBDevice {
```

Comment:
I agree, I'd like the ADBDevice class to be an actual Interface that USBDevice and TCPDevice implement.